src/daft-core/src/array/ops/comparison.rs

```diff
@@ -794,6 +799,56 @@ impl DaftLogical<&BooleanArray> for BooleanArray {
     }
 }
 
+
+// Implementaing Daft Logic for Numeric values
+macro_rules! impl_daft_logical_for_integers {
```

Comment:
You shouldn't need to use a macro to implement the `DaftLogical` trait, you can instead use generics:
```
impl<T> DaftLogical<&DataArray<T>> for DataArray<T>
where
    T: DaftIntegerType,
    <T as DaftNumericType>::Native: Ord,
{
    type Output = DaftResult<Series>;

    fn and(&self, rhs: &DataArray<T>) -> Self::Output {
        todo!("and")
    }

    fn or(&self, rhs: &DataArray<T>) -> Self::Output {
        todo!("or")
    }

    fn xor(&self, rhs: &DataArray<T>) -> Self::Output {
        todo!("xor")
    }
}
```

Reply:
@colin-ho  : Thanks for the quick response. I'm currently working on the changes you suggested. Will come back if i get stuck.